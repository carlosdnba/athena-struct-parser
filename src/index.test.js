const parser = require('./')

describe('index', () => {
  test('parses struct', () => {
    expect(parser('{date=2017-12-11T14:30:00-0700, primitive_null=null, primitive_bool=false, date2=2017-12-11T14:00:00-0700, id=76033946, list=[{string_in_list=accepted, null_in_list=null, id=392440526, email=user.name@7x7.com}, {string_in_list=not_responded, null_in_list=null, id=392440522, email=matt.damon@7x7.com}, {string_in_list=not_responded, null_in_list=null, id=392440525, user_id=165097, email=huh123123@7x7.com}, {string_in_list=accepted, null_in_list=null, id=392440527, email=rahi.sharma@7x7.com}, {string_in_list=not_responded, null_in_list=null, id=392440527, user_id=116927, email=uuhhh.a@7x7.com}, {string_in_list=accepted, null_in_list=null, id=392440523, email=other.name@7x7.com}], email=7x7.qerqwerqewrqwerqwerqw@qwerqwr.qwer.com, string_with_comma=Foo:bar bar, baz bazmeh, creator=joe.fragrant@7x7.com}')).toMatchSnapshot()
  })

  test('parses struct with whitespace', () => {
    expect(parser('{key=value,    some=email@some.com,     someStruct={a=b}    }')).toMatchSnapshot()
  })

  test('parses given list of boolean primitives', () => {
    expect(parser('[false, true, false, true]')).toMatchSnapshot()
  })

  test('parses given list with nulls', () => {
    expect(parser('[something, bar, baz, null, meh]')).toMatchSnapshot()
  })

  test('parses nested arrays', () => {
    expect(parser('[[a, b], [v, b]]')).toMatchSnapshot()
  })
})
