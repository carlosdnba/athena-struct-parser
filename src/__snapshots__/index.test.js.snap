// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index parses given list of boolean primitives 1`] = `
Array [
  false,
  true,
  false,
  true,
]
`;

exports[`index parses given list with nulls 1`] = `
Array [
  "something",
  "bar",
  "baz",
  null,
  "meh",
]
`;

exports[`index parses nested arrays 1`] = `
Array [
  Array [
    "a",
    "b",
  ],
  Array [
    "v",
    "b",
  ],
]
`;

exports[`index parses struct 1`] = `
Object {
  "creator": "joe.fragrant@7x7.com",
  "date": "2017-12-11T14:30:00-0700",
  "date2": "2017-12-11T14:00:00-0700",
  "email": "7x7.qerqwerqewrqwerqwerqw@qwerqwr.qwer.com",
  "id": "76033946",
  "list": Array [
    Object {
      "email": "user.name@7x7.com",
      "id": "392440526",
      "null_in_list": null,
      "string_in_list": "accepted",
    },
    Object {
      "email": "matt.damon@7x7.com",
      "id": "392440522",
      "null_in_list": null,
      "string_in_list": "not_responded",
    },
    Object {
      "email": "huh123123@7x7.com",
      "id": "392440525",
      "null_in_list": null,
      "string_in_list": "not_responded",
      "user_id": "165097",
    },
    Object {
      "email": "rahi.sharma@7x7.com",
      "id": "392440527",
      "null_in_list": null,
      "string_in_list": "accepted",
    },
    Object {
      "email": "uuhhh.a@7x7.com",
      "id": "392440527",
      "null_in_list": null,
      "string_in_list": "not_responded",
      "user_id": "116927",
    },
    Object {
      "email": "other.name@7x7.com",
      "id": "392440523",
      "null_in_list": null,
      "string_in_list": "accepted",
    },
  ],
  "primitive_bool": false,
  "primitive_null": null,
  "string_with_comma": "Foo:bar bar, baz bazmeh",
}
`;

exports[`index parses struct with whitespace 1`] = `
Object {
  "key": "value",
  "some": "email@some.com",
  "someStruct": Object {
    "a": "b",
  },
}
`;
